packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -321,10 +321,20 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       }
 
       // Restore the original state of the work-in-progress
+      if (stashedWorkInProgressProperties === null) {
+        // This should never happen. Don't throw because this code is DEV-only.
```

Comment:
I think this shouldn't happen in practice because we always assign them before performing work.
If it does we'll want to know.